doc/user-manual/tools/interface-files.rst

```diff
@@ -0,0 +1,47 @@
+.. _interface-files:
+
+********************
+Interface files
+********************
+
+.. note::
+   This is a stub. Contributions, additions and corrections are greatly
+   appreciated.
+
+Probably you already noticed that when you save an agda file, another file
+with the same name and extension .agdai is automatically created. This .agdai
+file is what we call an **interface file** that stores a compressed form of
+the type-checked version of the code that lives in the corresponding .agda file.
+
+Interface files store the results from the type-checking process and resolve
```

Comment:
I suggest the following more accurate version:

Interface files store the results from the type-checking process.  These results include: 
- A translation of pattern-matching definitions to case trees (this translation speeds up
computation).
- The resolution of all implicit arguments (unless you use the flag ``--allow-unsolved-metas``).

Reply:
Updated.